### PR TITLE
*: create project governance

### PR DESCRIPTION
**Pull request by spzala**
Create project governance.

### CODE REVIEW

**Comment by spzala** [comment]
@gyuho please take a look. If the flow and content looks good to you, I would request others to review. Thanks!

File: GOVERNANCE.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+supermajority of  maintainers. Likewise, maintainers can be removed by a supermajority
+of the maintainers and moved to emeritus status.
+
+Life priorities, interests, and passions can change. If a maintainer needs to step
+down, inform other maintainers about this intention, and if possible, help find someone
+to pick up the related work. At the very least, ensure the related work can be continued.
+Afterward, create a pull request to remove yourself from the [MAINTAINERS](./MAINTAINERS)
+file.
+
+## Approvers
```
Comment (jingyih) [inline GOVERNANCE.md]: Are approvers and reviewers refer to the list in OWNERS file?

Comment (spzala) [reply GOVERNANCE.md]: @jingyih thank you and yes, that's true. Good idea to provide link to the OWNERS file while mentioning them if that's what you are thinking! Also, most other projects usually have one file - MAINTAINERS or OWNERS - with maintainers/approvers/reviewers listed separately underneath. So I am up for merging them like moving maintainers in the OWNER file or other way.

Comment (jingyih) [reply GOVERNANCE.md]: I agree, having both files is also confusing to me, which is why I asked the question initially.

Comment (gyuho) [reply GOVERNANCE.md]: Yeah, merging into one sound better.

File: GOVERNANCE.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+to `etcd-maintainers@googlegroups.com`.
+
+## Conflict resolution
+
+In general, we prefer that technical issues and maintainer membership are amicably
+worked out between the persons involved. However, any technical dispute that has
+reached an impasse with a subset of the community, any contributor may open a GitHub
+issue or PR or send an email to `etcd-maintainers@googlegroups.com`. If the
+maintainers themselves cannot decide an issue, the issue will be resolved by a
+supermajority of the  maintainers.
```
Comment (gyuho) [inline GOVERNANCE.md]: `s/the  maintainers./the maintainers./`?

File: GOVERNANCE.md (hunk 1 above)
Comment (spzala) [reply GOVERNANCE.md]: OK, sounds great!

**Comment by spzala** [comment]
@gyuho @jingyih I am going with maintainers file considering its format is more readable. Other projects like Prometheus, Containerd, Fluentd also use maintainers file. Also, cleaned up the content of owners file a bit. Hope makes sense. Thanks!

**Comment by jingyih** [comment]
> @gyuho @jingyih I am going with maintainers file considering its format is more readable. Other projects like Prometheus, Containerd, Fluentd also use maintainers file. Also, cleaned up the content of owners file a bit. Hope makes sense. Thanks!

Thanks! The new maintainers file looks consistent with the governance doc.

**Comment by gyuho** [comment]
@spzala Let's highlight this in changelog as well. thx

**Comment by spzala** [comment]
Thanks @gyuho @jingyih !! Let's wait for rest of this week before we merge if we get any other comments? /cc @philips @xiang90 @jpbetz @hexfusion

**Comment by philips** [comment]
lgtm.

One discussion: Was there a discussion on the concept of approvers? Have we needed this additional abstraction in the past? And did Fanmin and Anthony agree to be approvers?

**Comment by spzala** [comment]
> lgtm.
> 
> One discussion: Was there a discussion on the concept of approvers? Have we needed this additional abstraction in the past? And did Fanmin and Anthony agree to be approvers?

@philips thanks, and good points for discussion something I also thought myself too but I decided to kept the roles the same way except removing overlapping of members between roles. Now that you mentioned, I will reach out to Fanmin and Anthony for their interest. Also, though current roles provides good distinction, most projects only have two (maintainers/approvers and reviewers). I personally is totally good in keeping only two roles (maintainers and approvers/reviewers) if everyone is fine with it.

**Comment by philips** [comment]
In my mind we either:

1. Remove approvers completely.
2. Rename maintainers to approvers.

A reviewer is generally someone who is on the ladder towards maintainership
but isn't experienced/trusted enough to merge code themselves.

IMHO we should just remove approvers because the language of "maintainers"
has been with the project since day 1.

**Comment by spzala** [comment]
@gyuho @jingyih @philips @xiang90 please look at the updated changes. Also, I have informed Fanmin and Anthony about this PR. Thanks!

File: GOVERNANCE.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+### Package/feature maintainer
+
+- Ownership of test and debug failures in a pkg/feature
+- Resolution of bugs triaged to a package/feature
+- Regularly review pull requests to the pkg subsystem
+
+Contributors who are interested in becoming a maintainer, if performing these
+responsibilities, should discuss their interest with the existing maintainers. New
+maintainers must be nominated by an existing maintainer and must be elected by a
+supermajority of  maintainers. Likewise, maintainers can be removed by a supermajority
```
Comment (gyuho) [inline GOVERNANCE.md]: extra whitespace character :)

`s/supermajority of  maintainers/supermajority of maintainers/`

Comment (spzala) [reply GOVERNANCE.md]: Ahh :), good catch. Thanks @gyuho

**Comment by jingyih** [comment]
LGTM! Thanks!

Not sure why travis is complaining about the commit title...

**Comment by spzala** [comment]
> LGTM! Thanks!
> 
> Not sure why travis is complaining about the commit title...

Thanks @jingyih - interesting, let me update PR to address comment from @gyuho and be more explicit with commit title, however as you pointed out travis failure is strange.

**Comment by philips** [comment]
LGTM

**Pull request merged**